nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/DeleteMarkLogic.java

```diff
@@ -69,25 +79,19 @@ public void init(ProcessorInitializationContext context) {
 
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory)
-        throws ProcessException {
+            throws ProcessException {
         super.onTrigger(context, sessionFactory);
     }
 
-    /**
-     * Overrides the behavior in the parent class for how each batch of URIs should be processed.
-     *
-     * @param context
-     * @param session
-     * @return
-     */
     @Override
-    protected QueryBatchListener buildQueryBatchListener(final ProcessContext context, final ProcessSession session) {
+    protected QueryBatchListener buildQueryBatchListener(final ProcessContext context, final ProcessSession session, final boolean consistentSnapshot) {
         return new NiFiDeleteListener(session).onFailure((batch, throwable) -> {
-            synchronized (session) {
+            synchronized(session) {
                 getLogger().error("Error deleting batch", throwable);
+                FlowFile flowFile = session.create();
```

Comment:
This causes the existing test - `DeleteMarkLogicIT` - to fail because NiFi doesn't allow for the same FlowFile to be transferred multiple times. The FlowFile creation should still happen in the for loop for each URI, but it should be `FlowFile flowFile = session.create(incomingFlowFile)` - which requires modifying `buildQueryBatcherListener` so that `FlowFile incomingFlowFile` is passed in.